cmapi/integration_tests/ssh.py

```diff
@@ -0,0 +1,120 @@
+import concurrent.futures
+from contextlib import contextmanager
+import json
+from pathlib import Path
+from typing import Any, Callable
+from fabric import Connection
+
+
+class HostConfig:
+    """Keeps configuration for a host in the cluster"""
+    def __init__(self, name: str, private_ip: str, public_fqdn: str, key_file_path: Path, ssh_user: str) -> None:
+        self.name = name
+        self.private_ip = private_ip
+        self.public_fqdn = public_fqdn
+        self.key_file_path = key_file_path
+        self.ssh_user = ssh_user
+
+    def __repr__(self) -> str:
+        return f"HostConfig(name={self.name}, private_ip={self.private_ip}, public_fqdn={self.public_fqdn}"
+
+    @contextmanager
+    def ssh_connection(self):
```

Comment:
[minor]
Seems to be misleading naming of the class. <placeholder>Config can't do connections imo.